server/webapp/WEB-INF/rails.new/app/controllers/api_v1/version_infos_controller.rb

```diff
@@ -0,0 +1,53 @@
+##########################################################################
+# Copyright 2015 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  class VersionInfosController < ApiV1::BaseController
+    def update_server
+      result = HttpLocalizedOperationResult.new
+
+      render :nothing => true, :status => 400 and return if message_tampered?
```

Comment:
There's a `render_message` helper, I'd suggest sending a message back with a status code. The preferred way to write this is to —

```ruby
return ... if message_tampered?
```